pkg/server/commands/list_objects.go

```diff
@@ -95,76 +114,109 @@ func (q *ListObjectsQuery) evaluate(
 	}
 
 	handler := func() {
-		defer close(resultsChan)
-
-		ctx = typesystem.ContextWithTypesystem(ctx, typesys)
-		span.SetAttributes(attribute.Bool(listObjectsOptimizedKey, false))
-
-		err = q.performChecks(ctx, req, resultsChan, maxResults)
-		if err != nil {
-			resultsChan <- ListObjectsResult{Err: err}
-		}
-	}
-
-	hasTypeInfo, err := typesys.HasTypeInfo(targetObjectType, targetRelation)
-	if err != nil {
-		q.Logger.WarnWithContext(
-			ctx, fmt.Sprintf("failed to lookup type info for relation '%s'", targetRelation),
-			zap.String("store_id", req.GetStoreId()),
-			zap.String("object_type", targetObjectType),
-		)
-	}
-
-	containsIntersection, _ := typesys.RelationInvolvesIntersection(targetObjectType, targetRelation)
-	containsExclusion, _ := typesys.RelationInvolvesExclusion(targetObjectType, targetRelation)
-
-	// ConnectedObjects currently only supports models that do not include intersection and exclusion,
-	// and the model must include type info for ConnectedObjects to work.
-	if !containsIntersection && !containsExclusion && hasTypeInfo {
 		userObj, userRel := tuple.SplitObjectRelation(req.GetUser())
-
 		userObjType, userObjID := tuple.SplitObject(userObj)
 
-		var targetUserRef isUserRef
-		targetUserRef = &UserRefObject{
+		var sourceUserRef isUserRef
+		sourceUserRef = &UserRefObject{
 			Object: &openfgapb.Object{
 				Type: userObjType,
 				Id:   userObjID,
 			},
 		}
 
 		if tuple.IsTypedWildcard(userObj) {
-			targetUserRef = &UserRefTypedWildcard{Type: tuple.GetType(userObj)}
+			sourceUserRef = &UserRefTypedWildcard{Type: tuple.GetType(userObj)}
+
 		}
 
 		if userRel != "" {
-			targetUserRef = &UserRefObjectRelation{
+			sourceUserRef = &UserRefObjectRelation{
 				ObjectRelation: &openfgapb.ObjectRelation{
 					Object:   userObj,
 					Relation: userRel,
 				},
 			}
 		}
 
-		handler = func() {
-			defer close(resultsChan)
-
-			span.SetAttributes(attribute.Bool(listObjectsOptimizedKey, true))
-
-			connectObjCmd := &ConnectedObjectsCommand{
-				Datastore:        q.Datastore,
-				ResolveNodeLimit: q.ResolveNodeLimit,
-				Limit:            maxResults,
-			}
```

Comment:
Why did we remove this and why are we passing this on every test? Same comment for `CheckResolver`

This file should build both internally because it absolutely needs those two things. Clients shouldn't pass them in. See https://github.com/openfga/openfga/issues/619

Reply:
How does it look to you now @miparnisari ? I rewired it up.